src/r_builtins/paste.rs

```diff
@@ -0,0 +1,222 @@
+use crate::ast::*;
+use crate::error::*;
+use crate::lang::*;
+use crate::r_builtins::builtins::force_closures;
+use crate::r_vector::vectors::*;
+
+pub fn primitive_paste(args: ExprList, env: &mut Environment) -> EvalResult {
+    let R::List(vals) = env.eval_list(args)? else {
+        unreachable!()
+    };
+
+    let vals = force_closures(vals);
+
+    let mut vec_s_vec: Vec<Vec<String>> = vec![];
+    let mut sep = String::from(" ");
+    let mut collapse = String::new();
+    let mut max_len: usize = 0;
+
+    // Extract sep and collapse parameters, leave the rest for processing
+    for (k, v) in vals {
+        let k_clone = k.clone().unwrap_or("".to_string());
+
+        match k_clone.as_str() {
+            "sep" => {
+                sep = match v {
+                    // We need to check whether the supplied sep value is a
+                    // character. R does not accept non-character values. If
+                    // we use R::Vector(v) => R::Vector(v.as_character()) then
+                    // this will coerce non-valid sep value to a character
+                    R::Vector(Vector::Character(v)) => v.get(0).unwrap().clone().to_string(),
+                    _ => {
+                        return Err(RSignal::Error(RError::Other(
+                            "sep parameter must be a character string!".to_string(),
+                        )))
+                    }
+                }
+            }
+
+            "collapse" => {
+                collapse = match v {
+                    R::Null => continue,
+                    R::Vector(Vector::Character(v)) => v.get(0).unwrap().clone().to_string(),
+                    _ => {
+                        return Err(RSignal::Error(RError::Other(
+                            "collapse parameter must be NULL or a character string!".to_string(),
+                        )))
+                    }
+                }
+            }
+
+            _ => match v {
+                R::List(_) => {
+                    return Err(RSignal::Error(RError::Other(
+                        "list is not supported in paste() yet!".to_string(),
+                    )))
+                }
+
+                R::Null => continue,
+
+                _ => {
+                    // Leave the rest for processing. Coerce everything into
+                    // character.
+                    let s_vec = v.clone().as_character().unwrap().get_vec_string();
+                    vec_s_vec.push(s_vec.clone());
+
+                    // Need the length of longest vector to create an empty
+                    // vector that others will go through and re-cycle values as
+                    // needed
+                    if s_vec.len() > max_len {
+                        max_len = s_vec.len()
+                    }
+                }
+            },
+        }
+    }
```

Comment:
Fully agreed, thanks so much. I was afraid of using the `remove` since its documentation says:
```
Note: Because this shifts over the remaining elements, it has a worst-case performance of O(n).
```

But I guess we don't need to worry too much about this since most of users just supply few inputs.